### PR TITLE
[Doris on ES] fix bug of issue 3287

### PR DESCRIPTION
**Pull request by BabySid**
#3287

### CODE REVIEW

File: be/src/exec/es/es_scroll_parser.cpp [hunk 1]
```diff
@@ -129,14 +140,13 @@ static const string ERROR_COL_DATA_IS_ARRAY = "Data source returned an array for
...
 static Status get_int_value(const rapidjson::Value &col, PrimitiveType type, void* slot, bool pure_doc_value) {
-
-
     if (col.IsNumber()) {
         *reinterpret_cast<T*>(slot) = (T)(sizeof(T) < 8 ? col.GetInt() : col.GetInt64());
         return Status::OK();
     }
 
     if (pure_doc_value && col.IsArray()) {
+	RETURN_ERROR_IF_COL_IS_NOT_NUMBER(col, type);
```
Comment (wuyunfeng) [inline be/src/exec/es/es_scroll_parser.cpp:149]: Can you delete the extra leading space?

File: be/src/exec/es/es_scroll_parser.cpp [hunk 2]
```diff
@@ -129,14 +140,13 @@ static const string ERROR_COL_DATA_IS_ARRAY = "Data source returned an array for
...
 static Status get_int_value(const rapidjson::Value &col, PrimitiveType type, void* slot, bool pure_doc_value) {
-
-
     if (col.IsNumber()) {
         *reinterpret_cast<T*>(slot) = (T)(sizeof(T) < 8 ? col.GetInt() : col.GetInt64());
         return Status::OK();
     }
 
     if (pure_doc_value && col.IsArray()) {
+        RETURN_ERROR_IF_COL_IS_NOT_NUMBER(col, type);
```
Comment (wuyunfeng) [inline be/src/exec/es/es_scroll_parser.cpp:149]: col is an array

**Review by wuyunfeng — APPROVED** [review]
LGTM @blackfox1983 Thanks

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**